ykrt/src/compile/jitc_yk/codegen/reg_alloc/mod.rs

```diff
@@ -31,6 +31,7 @@ pub(crate) enum LocalAlloc {
     ///
     /// FIXME: unimplemented.
     Register,
+    Const(u64),
```

Comment:
I think this needs to be `ConstInt` as we'll have other other `Const` things (e.g. pointers) later.

Reply:
Are you fine with just changing this to `ConstInt` and not worry about the different sizes for now?

Reply:
I'm fine fleshing it out later, I assume we have an `assert` (or similar) somewhere that means we can only get to this point for `u64` sized things? What we don't want to do is compile code that silently overwrites (say) two adjacent `u8`s with a `u64`.

Reply:
We don't. I don't want to add an `assert` that will likely fail lua tests. We are doing the same with loading an 8bit local. It just gets zero extended to the full register. So I think writing the whole u64 to the register is no different.

Reply:
> So I think writing the whole u64 to the register is no different.

What happens when we deopt a heap value?

Reply:
If that's the case why do we need to worry about the size of things in a guard?

Reply:
I'm not sure what you mean. I don't think that's something we worry about. This discussion was specifically about this line https://github.com/ykjit/yk/pull/1247/files#diff-150b793f9518b19505892d1496e8641dc1de27061360bfb755cb1c85159f3eaeR910. Which happens whenever we load a local that is actually a `ProxyConst`.

Reply:
Resolved offline.